contracts/DMMPoolV2.sol

```diff
@@ -0,0 +1,378 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "./libraries/MathExt.sol";
+import "./libraries/FeeFomula.sol";
+import "./libraries/ERC20Permit.sol";
+
+import "./interfaces/IDMMFactory.sol";
+import "./interfaces/IDMMCallee.sol";
+import "./interfaces/IDMMPool.sol";
+import "./interfaces/IERC20Metadata.sol";
+
+contract DMMPoolV2 is IDMMPool, ERC20Permit, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 internal constant MAX_UINT112 = 2**112 - 1;
+    uint256 internal constant BPS = 10000;
+
+    struct ReserveData {
+        uint256 reserve0;
+        uint256 reserve1;
+        uint256 vReserve0;
+        uint256 vReserve1; // only used when isAmpPool = true
+    }
+
+    uint256 public constant MINIMUM_LIQUIDITY = 10**3;
+    uint256 internal constant PRECISION = 10**18;
+
+    /// @dev To make etherscan auto-verify new pool, these variables are not immutable
+    IDMMFactory public override factory;
+    IERC20 public override token0;
+    IERC20 public override token1;
+
+    /// @dev uses single storage slot, accessible via getReservesData
+    uint112 internal reserve0;
+    uint112 internal reserve1;
+    uint32 public override ampBps;
+    /// @dev addition param only when amplification factor > 1
+    uint112 internal vReserve0;
+    uint112 internal vReserve1;
+
+    /// @dev vReserve0 * vReserve1, as of immediately after the most recent liquidity event
+    uint256 public override kLast;
+
+    /// @dev fixed factor for arbitrum
+    uint256 internal factorInPrecision;
+
+    event Mint(address indexed sender, uint256 amount0, uint256 amount1);
+    event Burn(address indexed sender, uint256 amount0, uint256 amount1, address indexed to);
+    event Swap(
+        address indexed sender,
+        uint256 amount0In,
+        uint256 amount1In,
+        uint256 amount0Out,
+        uint256 amount1Out,
+        address indexed to,
+        uint256 feeInPrecision
+    );
+    event Sync(uint256 vReserve0, uint256 vReserve1, uint256 reserve0, uint256 reserve1);
+
+    constructor() public ERC20Permit("KyberDMM LP", "DMM-LP", "1") {
```

Comment:
Please change to "KyberSwap LP", "KS-LP" instead
Should change all DMM -> KS, KyberDMM -> KyberSwap

Reply:
Done